src/solc_0.7/catalyst/AssetUpgrader.sol

```diff
@@ -156,7 +156,8 @@ contract AssetUpgrader is WithMetaTransaction, IAssetUpgrader {
     ) internal {
         require(assetId & IS_NFT != 0, "INVALID_NOT_NFT"); // Asset (ERC1155ERC721.sol) ensure NFT will return true here and non-NFT will return false
         _burnGems(from, gemIds);
-        _chargeSand(from, gemAdditionFee); // TODO per gems or flat fee ?
+        // @todo per gems or flat fee ?
```

Comment:
Can we get confirmation on this one way or the other?

Reply:
i just got answer on it
The plan is as follow:
- first support flat fee
- but later we might want to enable a variable fee based on the catalyst (added or present)
- or even have a variable fee based on the gems
The next 2 could be part of a new AssetUpgrader version deployed later

Reply:
Sounds good to me. We're set up already for the flat fee.